langchain-core/src/structured_query/tests/functional.test.ts

```diff
@@ -0,0 +1,310 @@
+import { test, expect } from "@jest/globals";
+import { Document } from "../../documents/document.js";
+import { FunctionalTranslator } from "../functional.js";
+import { Comparators, Visitor } from "../ir.js";
+
+describe("FunctionalTranslator", () => {
+  const translator = new FunctionalTranslator();
+
+  describe("getAllowedComparatorsForType", () => {
```

Comment:
As mentioned in the PR description, I only added the bare minimum tests needed to ensure the modification to the existing functionality did not break the `visitComparison` logic.